plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduClientSession.java

```diff
@@ -619,4 +621,11 @@ private KuduSplit toKuduSplit(KuduTableHandle tableHandle, KuduScanToken token,
             throw new TrinoException(GENERIC_INTERNAL_ERROR, e);
         }
     }
+
+    @PreDestroy
```

Comment:
Do I understand correctly that this will be called because this class is injected by Guice (hence lifecycle managed by airlift)?

Reply:
Yes. We don't have to wire the LifeCycleManager + call `addInstance` if the instance is created via Airlift injection/injector.